pkg/resources/create.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"k8s.io/client-go/dynamic"
+
+	"go.uber.org/zap"
+
+	"golang.org/x/xerrors"
+
+	triggersv1 "github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+	"github.com/tidwall/sjson"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	discoveryclient "k8s.io/client-go/discovery"
+)
+
+// FindAPIResource returns the APIResource definition using the discovery client c.
+func FindAPIResource(apiVersion, kind string, c discoveryclient.ServerResourcesInterface) (*metav1.APIResource, error) {
+	//resourceList, err := r.DiscoveryClient.ServerResourcesForGroupVersion(apiVersion)
+	resourceList, err := c.ServerResourcesForGroupVersion(apiVersion)
+	if err != nil {
+		return nil, xerrors.Errorf("Error getting kubernetes server resources for apiVersion %s: %s", apiVersion, err)
```

Comment:
While we are refactoring this, could we switch the `xerrors.Errorf` to `fmt.Errorf`? :smile:

Reply:
oh yeah good call!